### PR TITLE
Add Neovim Documentation

### PR DESCRIPTION
**Pull request by TaylorBeeston**
#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)

#### Description

- What does this PR change? Give us a brief description.

Adds links and information about using Neovim to develop with Astro!

### CODE REVIEW

File: src/pages/en/editor-setup.md [hunk 1]
```diff
@@ -28,7 +28,8 @@ To get started, install the [Astro VS Code Extension](https://marketplace.visual
...
 
 - [VS Code Extension on Open VSX](https://open-vsx.org/extension/astro-build/astro-vscode) <span style="margin: 0.25em;"><Badge variant="accent">Official</Badge></span> - The official Astro VS Code Extension, available on the Open VSX registry for open platforms like [VSCodium](https://vscodium.com/)
-- [Nova Extension](https://extensions.panic.com/extensions/sciencefidelity/sciencefidelity.astro/)<span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Syntax highlighting, IntelliSense, autocompletion for Astro
+- [Nova Extension](https://extensions.panic.com/extensions/sciencefidelity/sciencefidelity.astro/) <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Syntax highlighting, IntelliSense, autocompletion for Astro
+- Neovim [LSP](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#astro) and [TreeSitter](https://github.com/virchau13/tree-sitter-astro) Plugins <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Provides syntax highlighting/treesitter parsing, intellisense, and autocompletion for Astro inside of Neovim
```
Comment (sarah11918) [inline src/pages/en/editor-setup.md:32]:
```suggestion
- Neovim [LSP](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#astro) and [TreeSitter](https://github.com/virchau13/tree-sitter-astro) Plugins <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Syntax highlighting/treesitter parsing, IntelliSense, and autocompletion for Astro inside of Neovim
```

Thanks for this! I just made a couple of small edits to match the style of the entry above it.

**Comment by sarah11918** [comment]
Thanks for this @TaylorBeeston ! I'm glad you found a good first issue for your first contribution to the Docs repo! Please see the suggestion I made, and I look forward to adding this to the Editor Setup page!

File: src/pages/en/editor-setup.md (hunk 1 above)
Comment (Princesseuh) [reply src/pages/en/editor-setup.md:32]: Minor nitpick: auto-completion is included in the term `IntelliSense` (it's an umbrella term for many features)

Comment (sarah11918) [reply src/pages/en/editor-setup.md:32]: Thanks for this, @Princesseuh ! Then, I guess we should take it out of Nova as well. Would it sound "right" if we just removed autocompletion from both of those lists? Or, is there a different way you'd expect or prefer to see the description phrased?

Comment (Princesseuh) [reply src/pages/en/editor-setup.md:32]: I personally do not have a lot of attachment to the term IntelliSense in general to be honest, it's a cool marketing term Microsoft invented notably for Visual Studio (not-Code), but I'm not actually sure people say "Look at the IntelliSense!" nowadays

I feel like just writing `code completion` might be more informative?

Comment (sarah11918) [reply src/pages/en/editor-setup.md:32]: Thanks, both of you! Then @TaylorBeeston , would you like to reword both Neovim and Nova together, however you see fit (make them both match each other to the extent they should), and when @Princesseuh is happy with the wording, I'm happy!

Comment (TaylorBeeston) [reply src/pages/en/editor-setup.md:32]: Sure! I'll just switch it to code completion since that also works for me!

**Comment by sarah11918** [comment]
@Princesseuh - How does this look to you now?

File: src/pages/en/editor-setup.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ To get started, install the [Astro VS Code Extension](https://marketplace.visual
...
 
 - [VS Code Extension on Open VSX](https://open-vsx.org/extension/astro-build/astro-vscode) <span style="margin: 0.25em;"><Badge variant="accent">Official</Badge></span> - The official Astro VS Code Extension, available on the Open VSX registry for open platforms like [VSCodium](https://vscodium.com/)
-- [Nova Extension](https://extensions.panic.com/extensions/sciencefidelity/sciencefidelity.astro/)<span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Syntax highlighting, IntelliSense, autocompletion for Astro
+- [Nova Extension](https://extensions.panic.com/extensions/sciencefidelity/sciencefidelity.astro/) <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Syntax highlighting and code completion for Astro inside of Nova
+- Neovim [LSP](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#astro) and [TreeSitter](https://github.com/virchau13/tree-sitter-astro) Plugins <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Provides Syntax highlighting, treesitter parsing, and code completion for Astro inside of Neovim
```
Comment (Princesseuh) [inline src/pages/en/editor-setup.md:32]:
```suggestion
- [Nova Extension](https://extensions.panic.com/extensions/sciencefidelity/sciencefidelity.astro/) <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Provides syntax highlighting and code completion for Astro inside of Nova
- Neovim [LSP](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#astro) and [TreeSitter](https://github.com/virchau13/tree-sitter-astro) Plugins <span style="margin: 0.25em;"><Badge variant="neutral">Community</Badge></span> - Provides syntax highlighting, treesitter parsing, and code completion for Astro inside of Neovim
```

Comment (TaylorBeeston) [reply src/pages/en/editor-setup.md]: You're very welcome! =D

**Review by Princesseuh — APPROVED** [review]

**Comment by sarah11918** [comment]
Welcome to the Face Pile, @TaylorBeeston !  🥳

**Pull request merged**